nextcord/application_command.py

```diff
@@ -756,6 +756,8 @@ def from_callback(
                         isinstance(param.annotation, type)
                         and issubclass(param.annotation, Interaction)
                     )
+                    # will always be an interaction parameter (generic with the outermost type being Interaction)
+                    or typing_extensions.get_origin(param.annotation) is Interaction
```

Comment:
Good to consider, can simplify that to just checking subclass since Python considers classes to be subclasses of themselves.

Type-checkers won't really like that get_origin being used there since it can return None which is not a class type accepted by issubclass.